### PR TITLE
Update the storage mock so that copied objects can be read

### LINKED ISSUE #5705: FakeStorageRpc generation support for copy fails on read
Hi,
I'm trying to use LocalStorageHelper for a test.   I do this (scala)

```
    val request = CopyRequest
      .newBuilder
      .setSource(BlobId.of(sourceBucketName, sourceKey))
      .setTarget(BlobId.of(destinationBucketName, destinationKey))
      .build
    gcsClient.copy(request).getResult

    val obj =  gcsClient.get(BlobId.of(destinationBucketName, destinationKey))
    val content: Array[Byte] = obj.getContent(BlobSourceOption.generationMatch)

```

So that's a copy followed by a read action on the new file (gscClient is the `LocalStorageHelper` in this case).

However, during the copy action here:
https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/testing/FakeStorageRpc.java#L379

The target is stored in `metadata` without a generation, so the read throws a NPE attempting to perform a generationMatch.

I'm hoping to learn:

1) Do you agree this is a bug or at least "not optimal behavior" ?
2) Would you be open to a PR that does the following:
  a) on copy, checks if destination exists, if it exists, gets the existing generation + 1 
  b) if it doesn't exist gets 0
  c) uses that resolved generation in the target in the rewrite request target that is stored in the `metadata` in memory store.

The currently implemented generational match logic seems to be checking the destination:
https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/testing/FakeStorageRpc.java#L366

Since generations are always updated on modification, I think my proposed changes are still good, but this precondition check in the code felt a little awkward to me as I read it.

#### Environment details

1. Specify the API at the beginning of the title (for example, "BigQuery: ...")
   General, Core, and Other are also allowed as types: FakeStorageRpc
2. OS type and version: osx 10.14.5
3. Java version: `scala 2.11.12`
4. google-cloud-java version(s): `0.99.0-alpha`

#### Steps to reproduce

  1. use the LocalStorageHelper to copy a file
  2. use the LocalStorageHelper to read the file that was just copied

#### Code example

```
    val request = CopyRequest
      .newBuilder
      .setSource(BlobId.of(sourceBucketName, sourceKey))
      .setTarget(BlobId.of(destinationBucketName, destinationKey))
      .build
    gcsClient.copy(request).getResult

    val obj =  gcsClient.get(BlobId.of(destinationBucketName, destinationKey))
    val content: Array[Byte] = obj.getContent(BlobSourceOption.generationMatch)

```

#### Stack trace

```
java.lang.NullPointerException
	at com.google.cloud.storage.Blob$BlobSourceOption.toSourceOptions(Blob.java:99)
	at com.google.cloud.storage.Blob$BlobSourceOption.toSourceOptions(Blob.java:196)
	at com.google.cloud.storage.Blob.getContent(Blob.java:478)
	at com.datadog.spark.util.GCSUtil$.readObject(GCSUtil.scala:110)
```

#### Any additional information below	

Making sure to follow these steps will guarantee the quickest resolution possible.

Thanks!

### PR DESCRIPTION
**Pull request by philipsdoctor**
Fixes #5705 Allows objects copied with the storage mock to then be read.

### CODE REVIEW

File: google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/testing/LocalStorageHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            .newBuilder()
+            .setSource(BlobId.of(testBucket, sourceFile))
+            .setTarget(BlobId.of(testBucket, destinationFile))
+            .build();
+
+        return request;
+    }
+
+    @Test
+    public void testCopyCanBeRead() {
```
Comment (philipsdoctor) [inline google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/testing/LocalStorageHelperTest.java]: This test reproduces the issue and fails without my fix.  An object created by copy cannot be read without raising a NPE.  This issue is fixed by this PR.

**Comment by philipsdoctor** [comment]
Hi,
My organization has told me that I've been added to the CLA (I verified before I made a PR) the comment left is a little confusing, it states that the PR author (me) has signed the CLA but that one of the commits wasn't authored by me (they all were).  Maybe my git username/email need to be updated to clear this check?

**Comment by raphaelhamonnais** [comment]
Thx @philipsdoctor for proposing this change. I definitely support it.

**Comment by chingor13** [comment]
I'm not sure what's up with the googlebot as the commit history looks correct.

**Comment by philipsdoctor** [comment]
Looking into these test failures, apologies they all passed locally.  I'll get it fixed up.

**Comment by philipsdoctor** [comment]
@chingor13 
1) I removed the test in error, re-adding it, that was my mistake
2) this speech-it error is a little vague, I'm not clear why this was non-mergeable?

```
[07:44:22][ERROR] Cannot check out non-mergeable pull request as if merged.
Please attach a kokoro:force-run label to re-run the Kokoro build when this happens on subsequent commits.
```

**Pull request merged**